src/components/lib/cos_kernel_api.c

```diff
@@ -787,7 +787,7 @@ cos_tcap_transfer(arcvcap_t dst, tcap_t src, tcap_res_t res, tcap_prio_t prio)
 int
 cos_tcap_delegate(asndcap_t dst, tcap_t src, tcap_res_t res, tcap_prio_t prio, tcap_deleg_flags_t flags)
 {
-	u32_t yield     = flags & TCAP_DELEG_YIELD;
+	u32_t yield     = (flags && TCAP_DELEG_YIELD);
```

Comment:
Damn! I'm wrong here, yes. It must be `((flags & TCAP_DELEG_YIELD) != 0)`.
I understood the problem but I coded it wrong. We only have 1 bit that designates to yield or not but my code was wrong, that it was not differentiating between different flags (though that worked). But with `&`, yield will be 2 (0010) which will be lost in the next step when we pack it.

I'll fix it to be `((flags & TCAP_DELEG_YIELD) != 0)`